src/github.com/getlantern/flashlight/client/client.go

```diff
@@ -39,66 +39,90 @@ type Client struct {
 	priorCfg        *ClientConfig
 	priorTrustedCAs *x509.CertPool
 	cfgMutex        sync.RWMutex
-	balCh           chan *balancer.Balancer
-	balInitialized  bool
-	rpCh            chan *httputil.ReverseProxy
-	rpInitialized   bool
-	hqfd            fronted.Dialer
-	l               net.Listener
+
+	// Balanced CONNECT dialers.
+	balChMu        sync.Mutex
+	balCh          chan *balancer.Balancer
+	balInitialized bool
+
+	// Reverse HTTP proxies.
+	rpChMu        sync.Mutex
+	rpCh          chan *httputil.ReverseProxy
+	rpInitialized bool
+
+	hqfd fronted.Dialer
+	l    net.Listener
 }
 
-// ListenAndServe makes the client listen for HTTP connections.
-// onListening is a callback that gets invoked as soon as the server is
-// accepting TCP connections.
-func (client *Client) ListenAndServe(onListening func()) error {
-	l, err := net.Listen("tcp", client.Addr)
-	if err != nil {
-		return fmt.Errorf("Client unable to listen at %v: %v", client.Addr, err)
+// ListenAndServe makes the client listen for HTTP connections.  onListeningFn
+// is a callback that gets invoked as soon as the server is accepting TCP
+// connections.
+func (client *Client) ListenAndServe(onListeningFn func()) (err error) {
```

Comment:
Why make it a named return value if it's never returned by name?  Doesn't make the code any shorter either, because it makes you declare `l`.